charts/delphai-deployment/templates/mapping.yaml

```diff
@@ -38,14 +38,50 @@ spec:
   service: "{{ $.Release.Name }}:{{ $.Values.ports.httpgateway.port }}"
   {{ else }}
   {{- fail "No HTTP ports found to publish" -}}
-  {{ end }}
+  {{ end }} {{/* if $.Values.ingress.oAuth */}}
 
   allow_upgrade:
     - websocket
 
   timeout_ms: 60000
   idle_timeout_ms: 500000
 
+{{ if $.Values.ingress.publicApiEnabled }}
+---    
+apiVersion: getambassador.io/v2
+kind: Mapping
+
+metadata:
+  name: {{ $.Release.Name }}-api
+  namespace: {{ $.Release.Namespace }}
+  labels:
+    app.kubernetes.io/managed-by: {{ $.Release.Service | quote }}
+    app.kubernetes.io/instance: {{ $.Release.Name | quote }}
+    helm.sh/release: {{ $.Release.Name | quote }}
+    helm.sh/chart: "{{ $.Chart.Name }}-{{ $.Chart.Version | replace "+" "_" }}"
+
+spec:
+  host: "api.{{ $.clusterValues.baseDomain }}"
+  {{ if index $.Values.buildMetadata.labels "com.delphai.image.release" }}
+  prefix: {{ $.Values.ingress.publicApiPrefix | quote }}
+  {{ else }}
+  prefix: {{ printf "/%s%s" $.Release.Name $.Values.ingress.publicApiPrefix | quote }}
+  {{ end }} {{/* if index $.Values.buildMetadata.labels "com.delphai.image.release" */}}
+  prefix_regex: true
```

Comment:
Where is the regex itself? I expected to have `(/|$)` or so in `prefix:` above

Reply:
The regex itself is provided in the values. I'd rather have it defined in one place rather than count on the template concatenating strings to form a regex. This is more transparent.

I also added an example, commented-out value in the values file to better document the intention.

Reply:
I disagree, lets discuss in person